### PR TITLE
Make cursors iterable

### PR DESCRIPTION
**Pull request by kirjs**
This would allow using baobab collections in Angular 2 templates.

### CODE REVIEW

**Comment by kirjs** [comment]
I gave it a bit of thought, and now I don't think it's the exact behavior I would expect here. 

I think it would make sense to return cursors as the result of the iteration. 

@Yomguithereal Let me know what your thought are

**Comment by kirjs** [comment]
Updated to return cursors, makes way more sense this way.

**Comment by kirjs** [comment]
Unfortunately generators would require babel polyfill to be included, so I had to get rid of them.

**Comment by Yomguithereal** [comment]
Hello @kirjs,
This looks like a nice addition. Some questions for you:

* How does this code fare in a non-ES6 env (after build of course)?
* Can you show me an Angular2 example displaying what you want to achieve?

**Comment by kirjs** [comment]
Hey @Yomguithereal, 

1. In non ES6 environment there should be no differences after the compilation

2. I compiled a simple example of trying to iterate over baobab list cursor with ng-for.


[Current baobab version](http://jsbin.com/fomupiyebu/4/edit?js,console,output)
```error
"EXCEPTION: Cannot find a differ supporting object '[object Cursor]' in [null]"
```
[After the pull request](http://jsbin.com/xixaqanudo/1/edit?html,js,output)

**Comment by Yomguithereal** [comment]
Oops. Sorry, I typed the wrong number in my commit. I was referring to #385.

**Comment by Yomguithereal** [comment]
Ok. Seems fine. Can I just ask you to normalize the style of the PR to match the existing code base please?

File: src/cursor.js [hunk 1]
```diff
@@ -386,6 +386,38 @@ export default class Cursor extends Emitter {
   }
 
   /**
+   * Method used to allow iterating over cursors containing list-type data.
+   *
+   * e.g. for(let i of cursor) { ... }
+   *
+   * @returns {Object}        -  Each item sequentially
+   */
+  [Symbol.iterator](){
```
Comment (Yomguithereal) [inline src/cursor.js]: Can you put a space between parenthesis and bracket please :smile:?

File: src/cursor.js [hunk 2]
```diff
@@ -386,6 +386,38 @@ export default class Cursor extends Emitter {
...
+    const cursor = this;
+    const length = array.length;
+
+    return {
+      next: function(){
+        if(i < length){
+          return {
+            value: cursor.select(i++)
+          };
+        } else {
```
Comment (Yomguithereal) [inline src/cursor.js]: Can you write the `else` on a new line?

File: test/suites/cursor.js [hunk 3]
```diff
@@ -713,6 +713,20 @@ describe('Cursor API', function() {
       assert.strictEqual(listItem.rightmost().get(), 4);
     });
 
+    it('should be possible to iterate an array.', function() {
+      var result = [];
```
Comment (Yomguithereal) [inline test/suites/cursor.js]: `const` this one.

**Comment by kirjs** [comment]
rebased, applied style fixes and cleaned up commit history, ptal

**Comment by Yomguithereal** [comment]
Thanks @kirjs. Will merge your PR as soon as I come back from my trip abroad next week.

**Comment by Yomguithereal** [comment]
@kirjs: your PR is now merged :smile:. Thanks!

**Comment by Yomguithereal** [comment]
It seems the `Symbol.iterator` parts remains in the built files. This means babel's runtime is needed to run the lib. I need to find a way to avoid this. Any idea?

**Comment by Yomguithereal** [comment]
Yes, I think attaching the iterator to the prototype should be done only if we can assert that Symbol and Symbol.iterator exist in the current env.

**Pull request merged**